src/Development/IDE/Core/Shake.hs

```diff
@@ -778,16 +819,16 @@ useWithStaleFast' key file = do
   liftIO $ case r of
```

Comment:
Isn't the call to `getValues` made redundant by `lastValueIO`?

Reply:
We still need to know whether the key is the map so that we can decide to block or not?

Reply:
I don't see it, could you walk me through the cases?

Reply:
There are three cases:

1a) The value isn't in the store, and we can't (or fail to) compute a stale persistent value - we want to wait for the rule to compute a fresh value (matching the previous semantics)
1b) The value isn't in the store but we can compute a stale persistent value - return the value
2) The value is in the store - we can just return it. 

(sorry for the delay, hope you will be able to page this back in)